client/hwlib/src/models/devices.rs

```diff
@@ -20,83 +20,98 @@
 pub struct Audio {
+    pub identifier: String,
     pub model: String,
     pub vendor: String,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct Bios {
-    pub firmware_revision: String,
-    pub release_date: String,
-    pub revision: String,
+    pub firmware_revision: Option<String>,
+    pub release_date: Option<String>,
+    pub revision: Option<String>,
     pub vendor: String,
     pub version: String,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct Board {
     pub manufacturer: String,
     pub product_name: String,
     pub version: String,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct Chassis {
     pub chassis_type: String,
     pub manufacturer: String,
     pub sku: String,
     pub version: String,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
-#[allow(clippy::upper_case_acronyms)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct GPU {
-    pub family: String,
+    pub codename: Option<String>,
+    pub identifier: String,
     pub manufacturer: String,
+    pub status: Option<DeviceStatus>,
     pub version: String,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct NetworkAdapter {
     pub bus: String,
-    pub id: String,
+    pub capacity: i32,
+    pub identifier: String,
     pub model: String,
     pub vendor: String,
-    pub capacity: i32,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct PCIPeripheral {
     pub pci_id: String,
     pub name: String,
+    pub status: Option<DeviceStatus>,
     pub vendor: String,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct Processor {
-    pub family: String,
-    pub frequency: f64,
+    pub codename: String,
+    pub frequency: u64,
     pub manufacturer: String,
     pub version: String,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct USBPeripheral {
     pub usb_id: String,
     pub name: String,
+    pub status: Option<DeviceStatus>,
     pub vendor: String,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct VideoCapture {
+    pub identifier: String,
     pub model: String,
+    pub status: Option<DeviceStatus>,
     pub vendor: String,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
 pub struct WirelessAdapter {
+    pub identifier: String,
     pub model: String,
+    pub status: Option<DeviceStatus>,
     pub vendor: String,
 }
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub enum DeviceStatus {
+    KnownWorking,
+    KnownBreaking,
+    Unknown,
+}
```

Comment:
There's a slight oddity here, in the struct above, it's possible to encode `status: Some(Unknown)`, I'm not sure whether this is useful, would it be possible to remove the `Unknown` variant here and hence drop the `Known` prefixes? I mean like this
```rust
pub enum DeviceStatus {
    Working,
    Breaking,
}
```